packages/react-core/src/components/TextInputGroup/__tests__/TextInputGroupMain.test.tsx

```diff
@@ -0,0 +1,278 @@
+    render(<TextInputGroupMain>Test</TextInputGroupMain>);
+
+    const inputGroupMain = screen.getByText('Test');
+
+    expect(inputGroupMain).not.toHaveClass('pf-m-icon');
+  });
+
+  it('renders with class pf-m-icon when an icon prop is passed', () => {
+    render(<TextInputGroupMain icon="icon">Test</TextInputGroupMain>);
+
+    const inputGroupMain = screen.getByText('Test');
+
+    expect(inputGroupMain).toHaveClass('pf-m-icon');
+  });
+
+  it('renders with custom class names provided via prop', () => {
+    render(<TextInputGroupMain className="custom-class">Test</TextInputGroupMain>);
+
+    const inputGroupMain = screen.getByText('Test');
+
+    expect(inputGroupMain).toHaveClass('custom-class');
+  });
+
+  it('renders with class pf-c-text-input-group__text on the inputs parent', () => {
+    render(<TextInputGroupMain>Test</TextInputGroupMain>);
+
+    const inputGroupText = screen.getByRole('textbox').parentNode;
+
+    expect(inputGroupText).toHaveClass('pf-c-text-input-group__text');
+  });
+
+  it('renders the input with class pf-c-text-input-group__text-input', () => {
+    render(<TextInputGroupMain>Test</TextInputGroupMain>);
+
+    const input = screen.getByRole('textbox');
+
+    expect(input).toHaveClass('pf-c-text-input-group__text-input');
+  });
+
+  it('renders given input icon props', () => {
+    render(<TextInputGroupMain icon={<SearchIcon />} />);
+
+    expect(screen.getByRole('img', { hidden: true })).toBeInTheDocument();
+  });
+
+  it('passes the aria-label prop to the input', () => {
+    render(<TextInputGroupMain aria-label="Test label">Test</TextInputGroupMain>);
+
+    const input = screen.getByRole('textbox');
+
+    expect(input).toHaveAccessibleName('Test label');
+  });
+
+  it('passes the value prop to the input', () => {
+    render(<TextInputGroupMain value="value text">Test</TextInputGroupMain>);
+
+    const input = screen.getByRole('textbox');
+
+    expect(input).toHaveAttribute('value', 'value text');
+  });
+
+  it('passes the placeholder prop to the input', () => {
+    render(<TextInputGroupMain placeholder="placeholder text">Test</TextInputGroupMain>);
+
+    const input = screen.getByRole('textbox');
+
+    expect(input).toHaveAttribute('placeholder', 'placeholder text');
+  });
+
+  it('defaults to an input type of text', () => {
+    render(<TextInputGroupMain>Test</TextInputGroupMain>);
+
+    const input = screen.getByRole('textbox');
+
+    expect(input).toHaveAttribute('type', 'text');
+  });
+
+  it('updates the input type when a different type is passed', () => {
+    render(<TextInputGroupMain type="search">Test</TextInputGroupMain>);
+
+    const textInput = screen.queryByRole('textbox');
+    const searchInput = screen.getByRole('searchbox');
+
+    expect(textInput).not.toBeInTheDocument();
+    expect(searchInput).toBeVisible();
+  });
+
+  it('does not disable the input when TextInputGroupContext isDisabled is false', () => {
```

Comment:
I don't think this is necessary since there are tests verifying this context is passed appropriately in the TextInputGroup suite. Same with the other TextInputGroupContext below.

Reply:
I disagree, the tests concerning the context in TextInputGroup are ensuring that TextInputGroup is "sending the right messages", whereas the tests here are ensuring that TextInputGroupMain is properly getting data from the context and acting in the way we expect.

Reply:
Ok, I can see there's a chance this component doesn't read in the `isDisabled` and apply it to the input.